aws/src/main/java/org/apache/iceberg/aws/s3/S3InputStream.java

```diff
@@ -56,7 +63,15 @@ class S3InputStream extends SeekableInputStream implements RangeReadable {
   private final Counter readBytes;
   private final Counter readOperations;
 
-  private int skipSize = 1024 * 1024;
+  private int skipSize = 1024 * 1024;;
+  private RetryPolicy<Object> retryPolicy =
+      RetryPolicy.builder()
+          .handle(
+              ImmutableList.of(
+                  SSLException.class, SocketTimeoutException.class, SocketException.class))
+          .onFailure(failure -> openStream(true))
+          .withMaxRetries(3)
+          .build();
```

Comment:
I ended up removing RetryableInputStream for now; GCS is channel based and we should probably verify the Azure SDK doesn't already retry internally. We can always add the abstraction later if it ends up being needed but there's arguably not a compelling case for adding it up front when inlining Failsafe is simple. cc @danielcweeks